### PR TITLE
UofT-DSI | deep learning - Assignment 2

### PR DESCRIPTION
**Pull request by ShayanHodai**
## What changes are you trying to make? (e.g. Adding or removing code, refactoring existing code, adding reports)
I am trying to improve the model's performance by experimenting with different architectures, adding pre-trained embeddings, and tuning hyperparameters.

## What did you learn from the changes you have made?
I learned how to integrate pre-trained embeddings into my model and how changes in architecture and training parameters affect the model's performance.

## Was there another approach you were thinking about making? If so, what approach(es) were you thinking of?
I considered experimenting with adding dropout layers and adjusting the number of LSTM units, and training for more epochs.

## Were there any challenges? If so, what issue(s) did you face? How did you overcome it?
I had memory issues while working on the dataset on Colab and then I loaded 0.03 of the dataset. 

## How were these changes tested?
The changes were tested by training the model and observing its loss and accuracy on both training and validation data.

## A reference to a related issue in your repository (if applicable)
N/A

## Checklist
- [ x] I can confirm that my changes are working as intended

### CODE REVIEW

**Comment by ananyajha2000** [comment]
A2 complete
